lib/elixir/lib/protocol.ex

```diff
@@ -635,7 +635,8 @@ defmodule Protocol do
 
   ## Helpers
 
-  defp builtin do
+  @doc false
+  def __builtin__ do
```

Comment:
This seems to be required as long as we are not scoping calls to the module in which they occurred (which would be pretty difficult, afaik). Otherwise, there's no way to tell if the module in `{MyProtocol.BitString, :__impl__, 1}` is one of the "builtin" impls or not.